### PR TITLE
[JENKINS-39521] Force text plain onto log

### PR DESCRIPTION
**Pull request by imeredith**
# Description
Test included.

See [JENKINS-39521](https://issues.jenkins-ci.org/browse/JENKINS-39521).

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [ ] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@reviewbybees

### CODE REVIEW

File: blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/LogResource.java [hunk 1]
```diff
@@ -44,16 +44,9 @@ private void writeLog(StaplerRequest req, StaplerResponse rsp, AcceptHeader acce
...
-                case "text/html":
-                    rsp.setContentType("text/html;charset=UTF-8");
-                    rsp.setStatus(HttpServletResponse.SC_OK);
-                    req.setAttribute("html", Boolean.valueOf(true));
-                    break;
-                case "text/plain":
-                    rsp.setContentType("text/plain;charset=UTF-8");
-                    rsp.setStatus(HttpServletResponse.SC_OK);
-            }
+            rsp.setContentType("text/plain;charset=UTF-8");
```
Comment (michaelneale) [inline blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/LogResource.java]: Is this used by the step viewer or only the full screen log view?

File: blueocean-pipeline-api-impl/src/test/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineApiTest.java [hunk 2]
```diff
@@ -200,7 +200,7 @@ public void getPipelineJobRunsLogTest() throws Exception {
         WorkflowRun b1 = job1.scheduleBuild2(0).get();
         j.assertBuildStatusSuccess(b1);
 
-        HttpResponse<String> response = get("/organizations/jenkins/pipelines/pipeline1/runs/"+b1.getId()+"/log?start=0", 200,HttpResponse.class);
+        HttpResponse<String> response = get("/organizations/jenkins/pipelines/pipeline1/runs/"+b1.getId()+"/log?start=0", 200,"text/html",HttpResponse.class);
```
Comment (michaelneale) [inline blueocean-pipeline-api-impl/src/test/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineApiTest.java]: this says text/html but below says text/plain??

**Comment by michaelneale** [comment]
confused by this one (see comments)

File: blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/LogResource.java (hunk 1 above)
Comment (imeredith) [inline blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/LogResource.java]: Its the same code for both, we need to make sure steps are fine with it being text/plain

File: blueocean-pipeline-api-impl/src/test/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineApiTest.java (hunk 2 above)
Comment (imeredith) [inline blueocean-pipeline-api-impl/src/test/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineApiTest.java]: Sure, Its a test that requests a text/html just like a browser might but gets back a text/plain.

File: blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/LogResource.java (hunk 1 above)
Comment (imeredith) [inline blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/LogResource.java]: Steps look fine to me.

**Comment by michaelneale** [comment]
ok :bee: from me - I think the explanations make sense.

**Pull request merged**